ush/python/pygfs/jedi/jedi.py

```diff
@@ -152,119 +184,150 @@ def link_exe(task_config: AttrDict) -> None:
+            os.symlink(self.jedi_config.exe_src, self.jedi_config.exe)
 
+    @staticmethod
     @logit(logger)
-    def get_obs_dict(self, task_config: AttrDict) -> Dict[str, Any]:
-        """Compile a dictionary of observation files to copy
-
-        This method extracts 'observers' from the JEDI yaml and from that list, extracts a list of
-        observation files that are to be copied to the run directory
-        from the observation input directory
+    def get_jedi_dict(jedi_config_yaml: str, task_config: AttrDict, expected_keys: Optional[list] = None):
+        """Get dictionary of Jedi objects from YAML specifying their configuration dictionaries
 
         Parameters
         ----------
-        task_config: AttrDict
-            Attribute-dictionary of all configuration variables associated with a GDAS task.
+        jedi_config_yaml : str
+            path to YAML specifying configuration dictionaries for Jedi objects
+        task_config : str
+            attribute-dictionary of all configuration variables associated with a GDAS task
+
 
         Returns
         ----------
-        obs_dict: Dict
-            a dictionary containing the list of observation files to copy for FileHandler
+        None
         """
 
-        observations = find_value_in_nested_dict(self.config, 'observations')
+        # Initialize dictionary of Jedi objects
+        jedi_dict = AttrDict()
 
-        copylist = []
-        for ob in observations['observers']:
-            obfile = ob['obs space']['obsdatain']['engine']['obsfile']
-            basename = os.path.basename(obfile)
-            copylist.append([os.path.join(task_config.COM_OBS, basename), obfile])
-        obs_dict = {
-            'mkdir': [os.path.join(task_config.DATA, 'obs')],
-            'copy': copylist
-        }
-        return obs_dict
+        # Parse J2-YAML file for dictionary of JEDI configuration dictionaries
+        jedi_config_dict = parse_j2yaml(jedi_config_yaml, task_config)
 
-    @logit(logger)
-    def get_bias_dict(self, task_config: AttrDict, bias_file) -> Dict[str, Any]:
-        """Compile a dictionary of observation files to copy
+        # Loop through dictionary of Jedi configuration dictionaries
+        for yaml_name in jedi_config_dict:
+            # Make sure all required keys present or set to None
+            jedi_config_dict[yaml_name]['yaml_name'] = yaml_name
+            for key in jedi_key_list:
+                if key not in jedi_config_dict[yaml_name]:
+                    jedi_config_dict[yaml_name][key] = None
+
+            # Construct JEDI object
+            jedi_dict[yaml_name] = Jedi(jedi_config_dict[yaml_name])
+
+        # Make sure jedi_dict has the keys we expect
+        if expected_keys:
+            for jedi_dict_key in expected_keys:
+                if jedi_dict_key not in jedi_dict:
+                    raise Exception(f"FATAL ERROR: {jedi_dict_key} not present {jedi_config_yaml}")
```

Comment:
log this with `logger.execption` or `logger.error` and then raise the exception.